### PR TITLE
Swift 6 concurrency support

### PR DESCRIPTION
**Pull request by aashishpatil-g**
Mark as Sendable, 
Synchronized access in Dataconnect instance.

### CODE REVIEW

**Comment by ncooke3** [comment]
See https://github.com/firebase/firebase-ios-sdk/pull/13933 for how I worked around the `nonisolated` errors in old Xcodes.

File: Sources/DataConnect.swift
Comment (ncooke3) [inline Sources/DataConnect.swift:190]: This is unrelated to this PR and likely not very important in practice, but I had learned that Swift dictionaries strongly reference their values. So once an instance is added to the dictionary, it will never be deallocated. To work around this, you can box the value in a type that holds a weak reference to it. This way, the instance cache will not hold a strong reference and prevent it from be deallocated. Again though, likely not critical since there likely won't be many instances created and they likely won't be deallocated often/ever during lifetime of app.

Example of dictionary type
https://github.com/firebase/firebase-ios-sdk/blob/dd0a4c6cb84aace05e8e49469f0fb2c1f70fd155/FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatStorage.swift#L61-L63

Example of removing an instance from the dictionary upon deallocation
https://github.com/firebase/firebase-ios-sdk/blob/dd0a4c6cb84aace05e8e49469f0fb2c1f70fd155/FirebaseCore/Internal/Sources/HeartbeatLogging/HeartbeatStorage.swift#L110

Example of weak container type
https://github.com/firebase/firebase-ios-sdk/blob/dd0a4c6cb84aace05e8e49469f0fb2c1f70fd155/FirebaseCore/Internal/Sources/HeartbeatLogging/WeakContainer.swift

File: Sources/DataConnect.swift
Comment (ncooke3) [inline Sources/DataConnect.swift:121]: Since this class doesn't appear to retain auth or app check instance, maybe delay calling `Auth.auth(app: app)` and `AppCheck.appCheck(app: app)` in this scope, instead get them in the `GrpcClient` initializer. Perhaps this will prevent the need for the `nonisolated(unsafe) let appCheck = AppCheck.appCheck(app: app)` since the app check instance isn't being passed out of scope.

File: Sources/DataConnect.swift
Comment (aashishpatil-g) [reply Sources/DataConnect.swift:193]: Good to know. I'll track this for future though in reality we won't see many instances per SDK linkage. There should in theory be just one per connector. So even with multi connector apps, might not be more than single digits.

File: Sources/DataConnect.swift
Comment (ncooke3) [inline Sources/DataConnect.swift:19]: I'm gradually working through adding Swift 6 conformance to the SDKs, I'll prioritize addressing these ones so we can remove the preconcurrency attribute.

**Review by ncooke3 — APPROVED** [review]

File: Sources/DataConnect.swift
Comment (aashishpatil-g) [reply Sources/DataConnect.swift:19]: No worries - thanks!

File: Sources/DataConnect.swift
Comment (ncooke3) [reply Sources/DataConnect.swift:193]: +1, agreed

**Review by dconeybe — COMMENTED** [review]
Without knowing Swift, I'll still give my LGTM since it looks reasonable to me. Let me know if you need formal approval; otherwise, I'll defer to @ncooke3

File: Sources/DataConnect.swift
Comment (aashishpatil-g) [reply Sources/DataConnect.swift]: That change worked and resolved the nonisolated requirement in the outer class. Thanks!

**Pull request merged**
